numpy/version.py

```diff
@@ -20,4 +20,34 @@
 full_version = version
 short_version = version.split("+")[0]
 
-del get_versions, vinfo
+
+def make_version_info() -> tuple[int, int, int]:
+    """
+    We want to expose a numeric version tuple to make it easier for
+    dependencies to have conditional code without having to parse our version
+    string.
+
+    It can be tricky as 1) you don't want to compare string. 2) it can be
+    tempting to just split on dot and map convert to int, but that will fail on
+    rc, and others.
+
+    It is ok to drop all the non-numeric items as conditional code is unlikely
+    to rely on those values. We also don't add the non-numeric elements at the
+    end, as strings should anyway not be compared.
+
+    Matplotlib goes a bit further and make that a named tuple and include the
+    a/b/rc/... in the fourth place in the tuple, but this is generally
+    unnecessary. Though it could be added in a backward compatible manner.
```

Comment:
Not doing this is a bit questionable, and makes it fairly fragile to work with development versions. Is there a reason not to do what Matplotlib does here? So for release versions it'd be a length-3 tuple, and otherwise a length-4 tuple?

Reply:
You can't make it "just" a 4-tuple because:

```
>>> list(sorted(['rc', 'b', 'a', 'dev']))
['a', 'b', 'dev', 'rc']
```

And dev should be the highest. So you get the wrong order just with a tuple, and  you do need a custom object (like NumPyVersion, which I'm fine using). 

IMHO conditional checking of version number that contain a/b/rc/dev is an anti pattern, but I can see why you want it.


I don't like seeing 

```
np.__version_info__ < '1.22.0'
```

Because from an outside reader you don't know that `__version_info__` is a custom object that will parse `'1.22.0'` in its `__lt__`/`__gt__`/`__eq__`. And it makes it _easy_ for other project, or dev to think "oh it's ok to compare a string", and copy the bad practice. And the problem is that a lot of folks take numpy as example. 

Now my only goal here is to get a standard, hard to get wrong way of comparing versions, I'd like something that does not require super complex dependency or code (otherwise we should just rely on `from packaging.version import parse, Version` maybe? And a (int, int, int) seem like the simplest. I'm happy with more complicated though.

Reply:
> IMHO conditional checking of version number that contain a/b/rc/dev is an anti pattern, but I can see why you want it.

I am more worried about `__version_info__ < (1, 2, 0)` simply being incorrect for any `1.2.0.dev0` version.